pixeltable/globals.py

```diff
@@ -600,18 +601,32 @@ def create_dir(
         Create a directory and replace if it already exists:
 
         >>> pxt.create_dir('my_dir', if_exists='replace_force')
+
+        Create a subdirectory along with its ancestors:
+
+        >>> pxt.create_dir('parent1.parent2.sub_dir', create_parents=True)
     """
     path_obj = catalog.Path(path)
     cat = Catalog.get()
 
     with env.Env.get().begin_xact():
+        if create_parents:
+            # start walking from the root
+            ancestors = path_obj.ancestors()
+            root_path = catalog.Path('', empty_is_valid=True)
+            last_parent = cat.get_schema_object(str(root_path))
+            for ancestor in ancestors:
+                ancestor_obj = cat.get_schema_object(str(ancestor), expected=catalog.Dir)
+                last_parent = (
+                    catalog.Dir._create(last_parent._id, ancestor.name) if ancestor_obj is None else ancestor_obj
+                )
+            # get parent again
+            parent = cat.get_schema_object(str(path_obj.parent))
         if_exists_ = catalog.IfExistsParam.validated(if_exists, 'if_exists')
         existing = _handle_path_collision(path, catalog.Dir, False, if_exists_)
         if existing is not None:
             assert isinstance(existing, catalog.Dir)
             return existing
-
-        parent = cat.get_schema_object(str(path_obj.parent))
         assert parent is not None
```

Comment:
How does this work? It doesn't look like parent even gets set when !create_parents.

Reply:
yup interestingly all tests passed, adding coverage :)

Reply:
fixed the bug